### ISSUE TITLE
LP1995454: Graphics/*_driver_version_* failed

### ISSUE THREAD

**Issue by beliaev-maksim** [canonical/checkbox#59]
This issue was migrated from https://bugs.launchpad.net/plainbox-provider-checkbox/+bug/1995454

## Summary
| Status 	| Created on 	| Heat 	| Importance 	| Security related 	|
|--------	|------------	|------	|------------	|------------------	|
| Confirmed	| 2022-11-02 07:30:18	| 6	| Critical	| False  	|

## Description
[Summary]
Run Checkbox Graphic test item the following test were fail:
1.Graphics/1_driver_version_PCI_ID_0x4688
2.Graphics/2_driver_version_GA107BM_GeForce_RTX_3050_Ti_Mobie

[`Graphics/1_driver_version_PCI_ID_0x4688` Output]
error: list of process IDs must follow -p

Usage:
 ps [options]

 Try 'ps --help <simple|list|output|threads|misc|all>'
  or 'ps --help <s|l|o|t|m|a>'
 for additional help text.

For more details see ps(1).
Traceback (most recent call last):
  File "/tmp/nest-ynqwi8rq.1531dd90874d443ab9a3fbf7a63c668f7a0ab3881a634524bc7811bd83eed721/graphics_driver.py", line 399, in <module>
    sys.exit(main())
  File "/tmp/nest-ynqwi8rq.1531dd90874d443ab9a3fbf7a63c668f7a0ab3881a634524bc7811bd83eed721/graphics_driver.py", line 373, in main
    xorg_owner = check_output("ps -o user= -p $(pidof Xorg)",
  File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'ps -o user= -p $(pidof Xorg)' returned non-zero exit status 1.

[`Graphics/2_driver_version_GA107BM_GeForce_RTX_3050_Ti_Mobie` Output]
error: list of process IDs must follow -p

Usage:
 ps [options]

 Try 'ps --help <simple|list|output|threads|misc|all>'
  or 'ps --help <s|l|o|t|m|a>'
 for additional help text.

For more details see ps(1).
Traceback (most recent call last):
  File "/tmp/nest-ritxmv9m.6fe1458016043765a23331f4a48413f7f2bbb8cb9332037b39fb229ff44db35e/graphics_driver.py", line 399, in <module>
    sys.exit(main())
  File "/tmp/nest-ritxmv9m.6fe1458016043765a23331f4a48413f7f2bbb8cb9332037b39fb229ff44db35e/graphics_driver.py", line 373, in main
    xorg_owner = check_output("ps -o user= -p $(pidof Xorg)",
  File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'ps -o user= -p $(pidof Xorg)' returned non-zero exit status 1.

[Addtional Information]
Test Vault ID: 100505
image/manifest:dell-bto-jammy-jellyfish-gardevoir-X44-20220928-7.iso
Product Name:Galio
Bios Version:0.2.15
Kernel Version:6.0.0.1004-oem
CPU:Raptor Lake HX ES
GPU:Nvidia GN20-P1
Sku:All
Failure rate:2/2 units,4/4 times.

## Attachments
[Graphic.7z](https://api.launchpad.net/devel/bugs/1995454/+attachment/5628435/data)

Tags:
[]

**Comment by beliaev-maksim** [comment]
This thread was migrated from launchpad.net
##### https://launchpad.net/~huntu207 wrote on 2022-11-08 07:26:45:
This issue observed after checkbox resume session to run as root

**Comment by yphus** [comment]
Analysis: The pidof subshell command seems to fail to find the Xorg process. Either we enforce the protection around the subprocess call or we add proper support for 22.04/wayland.

**Comment by pseudocc** [comment]
`XDG_SESSION_TYPE` will be unset during suspend, and will be set to "wayland"/"x11" by the `graphical-session.target` service after the resume.

This job is called from
```bash
# shellcheck disable=SC1091
source graphics_env.sh {driver} {index}
if [[ $XDG_SESSION_TYPE == "wayland" ]]
then
  inxi_snapshot -Gazy
else
  graphics_driver.py
fi
```

**Comment by pieqq** [comment]
The fix provided in #523 should fix this issue. Closing it.
